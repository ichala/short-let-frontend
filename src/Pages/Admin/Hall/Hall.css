.btn-add-hall {
  max-width: 110px;
}
