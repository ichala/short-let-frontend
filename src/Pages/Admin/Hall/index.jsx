import { useEffect, useState } from 'react';
import { fetchHalls } from '../../../Api/admins/halls/api';
import Create from './components/create';
import HallsTable from './components/HallsTable';

const Hall = () => {
  const [halls, setHalls] = useState([]);

  // Fetch all halls when the the hall is changed or added
  const [changed, setChanged] = useState(false);

  useEffect(() => {
    fetchHalls().then((response) => {
      setHalls(response);
      setChanged(false);
    });
  }, [changed]);

  return (
    <div className="p-md-5">
      <div className="d-flex justify-content-between align-items-center mb-2">
        <h2>Halls list</h2>
        <Create setChanged={setChanged} />
      </div>
      <HallsTable halls={halls} setChanged={setChanged} />
    </div>
  );
};

export default Hall;
