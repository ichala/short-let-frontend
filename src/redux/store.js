import { configureStore } from '@reduxjs/toolkit';
import authSlices from './slices/auth';

export default configureStore({
  reducer: {
    user: authSlices,
  },
});
