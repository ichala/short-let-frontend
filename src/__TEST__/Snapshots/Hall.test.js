import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../redux/store';
import Hall from '../../Pages/Admin/Hall/index';

describe('Login', () => {
  it('should render the login page', () => {
    const app = renderer.create(
      <Provider store={store}>
        <Router>
          <Hall />
        </Router>
      </Provider>,
    );
    expect(app).toMatchSnapshot();
  });
});
