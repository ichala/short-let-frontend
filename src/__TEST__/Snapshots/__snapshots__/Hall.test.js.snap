// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the login page 1`] = `
<div
  className="p-md-5"
>
  <div
    className="d-flex justify-content-between align-items-center mb-2"
  >
    <h2>
      Halls list
    </h2>
    <div
      className="newhall"
    >
      <button
        className="btn btn-outline-success btn-add-hall"
        data-bs-target="#newhall"
        data-bs-toggle="modal"
        type="button"
      >
        <div
          className="d-flex justify-content-between align-items-center"
        >
          <span
            className="pb-1"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                }
              }
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              />
            </svg>
          </span>
          <span
            className="ms-1"
          >
            Add Hall
          </span>
        </div>
      </button>
      <div
        aria-hidden="true"
        aria-labelledby="staticBackdropLabel"
        className="modal fade modal-lg"
        data-bs-backdrop="static"
        data-bs-keyboard="false"
        id="newhall"
        tabIndex="-1"
      >
        <div
          className="modal-dialog"
        >
          <div
            className="modal-content"
          >
            <div
              className="modal-header"
            >
              <h5
                className="modal-title"
                id="staticBackdropLabel"
              >
                Add a new hall
              </h5>
              <button
                aria-label="Close"
                className="btn-close"
                data-bs-dismiss="modal"
                type="button"
              />
            </div>
            <div
              className="modal-body"
            >
              <form
                onSubmit={[Function]}
              >
                <div
                  className="form-group mb-2"
                >
                  <label
                    className="fomr-label w-100"
                    htmlFor="name"
                  >
                    Name:
                    <input
                      className="form-control"
                      id="name"
                      name="name"
                      onChange={[Function]}
                      placeholder="Hall name"
                      required={true}
                      type="text"
                      value=""
                    />
                  </label>
                </div>
                <div
                  className="form-group mb-2"
                >
                  <label
                    className="fomr-label w-100"
                    htmlFor="capacity"
                  >
                    Capacity:
                    <input
                      className="form-control"
                      id="capacity"
                      max={10000}
                      min={10}
                      name="capacity"
                      onChange={[Function]}
                      placeholder="Hall capacity"
                      required={true}
                      type="number"
                      value={0}
                    />
                  </label>
                </div>
                <div
                  className="form-group mb-2"
                >
                  <label
                    className="fomr-label w-100"
                    htmlFor="cost"
                  >
                    Cost:
                    <input
                      className="form-control"
                      id="cost"
                      name="cost"
                      onChange={[Function]}
                      placeholder="Hall cost"
                      required={true}
                      type="number"
                      value={0}
                    />
                  </label>
                </div>
                <div
                  className="form-group mb-2"
                >
                  <label
                    className="fomr-label w-100"
                    htmlFor="image"
                  >
                    Image:
                    <input
                      className="form-control"
                      id="image"
                      name="image"
                      onChange={[Function]}
                      placeholder="Hall image link"
                      required={true}
                      type="text"
                      value=""
                    />
                  </label>
                </div>
                <div
                  className="form-group mb-2"
                >
                  <label
                    className="fomr-label w-100"
                    htmlFor="description"
                  >
                    Description:
                    <textarea
                      className="form-control"
                      id="description"
                      minLength={10}
                      name="description"
                      onChange={[Function]}
                      placeholder="Hall description"
                      required={true}
                      value=""
                    />
                  </label>
                </div>
                <button
                  className="btn btn-success"
                  type="submit"
                >
                  Submit
                </button>
              </form>
            </div>
            <div
              className="modal-footer"
            >
              <button
                className="btn btn-danger"
                data-bs-dismiss="modal"
                type="button"
              >
                Close
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-dmRaPn gelpCx"
  >
    <div
      className="sc-fLlhyt ifOHjV"
    >
      <div
        className="sc-bczRLJ eSTlnH rdt_Table"
        disabled={false}
        role="table"
      >
        <div
          className="sc-ivTmOn fwKvpK"
        >
          <div
            style={
              Object {
                "padding": "24px",
              }
            }
          >
            There are no records to display
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
